src/core/evaluator.js

```diff
@@ -4257,10 +4255,21 @@ class PartialEvaluator {
           `The FontDescriptor's FontName is "${fontNameStr}" but ` +
             `should be the same as the Font's BaseFont "${baseFontStr}".`
         );
-        // Workaround for cases where e.g. fontNameStr = 'Arial' and
-        // baseFontStr = 'Arial,Bold' (needed when no font file is embedded).
-        if (fontNameStr && baseFontStr?.startsWith(fontNameStr)) {
-          fontName = baseFont;
+        if (fontNameStr && baseFontStr) {
+          // Workaround for cases where e.g. fontNameStr = 'Arial' and
+          // baseFontStr = 'Arial,Bold' (needed when no font file is embedded).
+          if (baseFontStr.startsWith(fontNameStr)) {
+            fontName = null;
+          }
+          // Workaround for cases where e.g. fontNameStr = 'wg09np' and
+          // baseFontStr = 'Wingdings-Regular' (fixes issue7454.pdf).
+          if (
```

Comment:
You could replace the `if` by a `else if`.

Reply:
You could merge the new if if the above one.

Reply:
Sure, I was just a little bit worried about overall readability by doing that.

Reply:
The meaning of the first two checks is pretty obvious and are here to allow the following ones, so from my pov it's ok.